pkg/upstream/cluster/loadbalancer.go

```diff
@@ -129,5 +131,58 @@ func (lb *roundRobinLoadBalancer) HostNum(metadata api.MetadataMatchCriteria) in
 	return len(lb.hosts.Hosts())
 }
 
+// leastActiveRequestLoadBalancer choose the host with the least active request
+type leastActiveRequestLoadBalancer struct {
+	hosts types.HostSet
+	rand  *rand.Rand
+}
+
+func newleastActiveRequestLoadBalancer(hosts types.HostSet) types.LoadBalancer {
+	return &leastActiveRequestLoadBalancer{
+		hosts: hosts,
+		rand:  rand.New(rand.NewSource(time.Now().UnixNano())),
+	}
+}
+
+func (lb *leastActiveRequestLoadBalancer) ChooseHost(context types.LoadBalancerContext) types.Host {
+	healthHosts := lb.hosts.HealthyHosts()
+	if len(healthHosts) == 0 {
+		return nil
+	}
+	// exactly one healthy host, return this host directly
+	if len(healthHosts) == 1 {
+		return healthHosts[0]
+	}
+	// The list of hosts having the same least active request value
+	candicate := make([]types.Host, 0, len(healthHosts))
+	// The least active request value of all hosts
+	leastActive := int64(math.MaxInt64)
+	for _, host := range healthHosts {
```

Comment:
If there are many hosts, it's slow.
We could begin host loop with a random index, and return it directly if the count is zero.

Reply:
return it directly if the count is zero is a good idea, But is it really necessary to loop hosts with a random index?

Reply:
done.

Reply:
such as RoundRobinLB，we first choose host from a random index.
I think that we could do it here.
```
r := random
for i := r; i < len(hosts); i ++ {
}
for i := 0; i < r; i ++ {
}
```